modules/javafx.graphics/src/main/java/com/sun/javafx/text/PrismTextLayout.java

```diff
@@ -452,13 +452,23 @@ public Hit getHitInfo(float x, float y) {
                 charIndex = run.getStart() + run.getOffsetAtX(x, trailing);
                 leading = (trailing[0] == 0);
 
-                // Calculate insertion index for surrogate pair text in TextFlow
-                // when leading is false
-                if (run.getTextSpan() != null) {
-                    String txt = run.getTextSpan().getText();
-                    int relIdx = charIndex - run.getStart();
-                    if (Character.isHighSurrogate(txt.charAt(relIdx)) && !leading) {
-                        insertionIndex = charIndex + 2;
+                insertionIndex = charIndex;
+                String txt = new String(getText());
```

Comment:
possible NPE: looks like getText() might return null, in which case this line will throw an NPE.

Reply:
Updated code